crates/optimism/evm/src/execute.rs

```diff
@@ -352,8 +352,10 @@ where
         // NOTE: we need to merge keep the reverts for the bundle retention
         self.state.merge_transitions(BundleRetention::Reverts);
 
+        let cache = core::mem::take(&mut self.state.cache);
```

Comment:
replace with self.state.take_cache() when https://github.com/bluealloy/revm/pull/1756 is merged